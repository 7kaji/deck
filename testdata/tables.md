# Table Test

Simple table with alignments:

| Name | Age | City |
|:-----|:---:|-----:|
| Alice | 25 | Tokyo |
| Bob | 30 | Osaka |

---

# Multiple Tables

First table:

| Item | Price |
|------|-------|
| Apple | $1.00 |
| Banana | $0.50 |

Second table:

| Language | Year |
|----------|------|
| Go | 2009 |
| Rust | 2010 |

---

# Table with Complex Content

| Feature | **Bold** | *Italic* | `Code` |
|---------|----------|----------|--------|
| Text | **Important** | *Emphasis* | `function()` |
| Links | [Google](https://google.com) | *Link* | `var x = 1` |