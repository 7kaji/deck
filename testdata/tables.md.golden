[
  {
    "layout": "",
    "titles": [
      "Table Test"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Simple table with alignments:"
              }
            ]
          }
        ]
      }
    ],
    "tables": [
      {
        "rows": [
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Name",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Age",
                    "bold": true
                  }
                ],
                "alignment": "CENTER",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "City",
                    "bold": true
                  }
                ],
                "alignment": "END",
                "is_header": true
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Alice"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "25"
                  }
                ],
                "alignment": "CENTER"
              },
              {
                "content": [
                  {
                    "value": "Tokyo"
                  }
                ],
                "alignment": "END"
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Bob"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "30"
                  }
                ],
                "alignment": "CENTER"
              },
              {
                "content": [
                  {
                    "value": "Osaka"
                  }
                ],
                "alignment": "END"
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Table Test"
      ]
    }
  },
  {
    "layout": "",
    "titles": [
      "Multiple Tables"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "First table:"
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Second table:"
              }
            ]
          }
        ]
      }
    ],
    "tables": [
      {
        "rows": [
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Item",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Price",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Apple"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "$1.00"
                  }
                ],
                "alignment": "START"
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Banana"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "$0.50"
                  }
                ],
                "alignment": "START"
              }
            ]
          }
        ]
      },
      {
        "rows": [
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Language",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Year",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Go"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "2009"
                  }
                ],
                "alignment": "START"
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Rust"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "2010"
                  }
                ],
                "alignment": "START"
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Multiple Tables"
      ]
    }
  },
  {
    "layout": "",
    "titles": [
      "Table with Complex Content"
    ],
    "tables": [
      {
        "rows": [
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Feature",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Bold",
                    "bold": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Italic",
                    "bold": true,
                    "italic": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              },
              {
                "content": [
                  {
                    "value": "Code",
                    "bold": true,
                    "code": true
                  }
                ],
                "alignment": "START",
                "is_header": true
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Text"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "Important",
                    "bold": true
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "Emphasis",
                    "italic": true
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "function()",
                    "code": true
                  }
                ],
                "alignment": "START"
              }
            ]
          },
          {
            "cells": [
              {
                "content": [
                  {
                    "value": "Links"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "Google",
                    "link": "https://google.com"
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "Link",
                    "italic": true
                  }
                ],
                "alignment": "START"
              },
              {
                "content": [
                  {
                    "value": "var x = 1",
                    "code": true
                  }
                ],
                "alignment": "START"
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Table with Complex Content"
      ]
    }
  }
]